src/Aspire.Hosting/ApplicationModel/ContainerExecutableResource.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Executable resource that runs in a container.
+/// </summary>
+internal class ContainerExecutableResource(string name, ContainerResource containerResource, string command, string? workingDirectory)
+    : Resource(name), IResourceWithEnvironment, IResourceWithArgs, IResourceWithEndpoints, IResourceWithWaitSupport
```

Comment:
@DeagleGross this merged in before I got a chance to fully review it - but one thing we should do here is ditch the `TargetContainerResource` property and make this resource implement the `IResourceWithParent` interface.

Reply:
We should also get rid of the `Args` property and instead make it work similar to the `ExecutableResource`. Because you are implementing `IResourceWithArgs` it means that this resource should be able to work with the `WithArgs` method - and it should honor all of the semantics around deferred evaluation etc.